api/ocm/tools/transfer/transferhandler/plugin/options.go

```diff
@@ -0,0 +1,178 @@
+package plugin
+
+import (
+	"encoding/json"
+
+	"github.com/mandelsoft/goutils/errors"
+
+	"ocm.software/ocm/api/ocm/tools/transfer/transferhandler"
+	"ocm.software/ocm/api/ocm/tools/transfer/transferhandler/standard"
+	"ocm.software/ocm/api/utils/runtime"
+)
+
+func init() {
+	transferhandler.RegisterHandler(100, &TransferOptionsCreator{})
+}
+
+type Options struct {
+	standard.Options
+	plugin  string
+	handler string
+	config  interface{}
+}
+
+var (
+	_ transferhandler.TransferOption = (*Options)(nil)
+	_ transferhandler.ConfigOption   = (*Options)(nil)
+
+	_ PluginNameOption            = (*Options)(nil)
+	_ TransferHandlerOption       = (*Options)(nil)
+	_ TransferHandlerConfigOption = (*Options)(nil)
+)
+
+type TransferOptionsCreator = transferhandler.SpecializedOptionsCreator[*Options, Options]
+
+func (o *Options) NewOptions() transferhandler.TransferHandlerOptions {
+	return &Options{}
+}
```

Comment:
So, the `transferhandler.SpecializedOptionsCreator[*Options, Options]` creates a "version" of `Option` that only provides the following interface:
```go
interface {
  NewOptions() transferhandler.TransferHandlerOptions
}
```
What's the purpose of this, seeing that Options implements the interface itself?

Reply:
It is the base implementation for a specialized transfer handler, which always should be able provide an options object for its own configuration.

This is used by the heuristic, which figures out the best matching handler implementation for a given set of options.
Once determined, the final options implementation is provided by the selected handler.

Reply:
The only doc , so far, is the plugin reference docu generated from the ppi command framework, which describes the plugin interface.

For the ppi itself, there is only the go doc. and examples, but no closed document describing how to use ppi